bash_completion

```diff
@@ -451,15 +492,26 @@ _comp_compgen()
         # in bash < 5.0, where unquoted $* in conditional command did not honor
         # IFS, was a bug.
         local IFS=$' \t\n'
-        [[ $* == *\$[0-9]* || $* == *\$\{[0-9]* ]]
-    then
         # Note: extglob *\$?(\{)[0-9]* can be extremely slow when the string
         # "${*:2:_nopt}" becomes longer, so we test \$[0-9] and \$\{[0-9]
         # separately.
+        [[ $* == *\$[0-9]* || $* == *\$\{[0-9]* ]]
+    then
         printf 'bash_completion: %s: positional parameter $1, $2, ... do not work inside this function.\n' "$FUNCNAME" >&2
         return 2
     fi
 
+    if [[ $1 != -* ]] && declare -f "_comp_compgen_$1" &>/dev/null; then
```

Comment:
Would it not be better to error out if the target function does not exist, rather than fall through?

Reply:
Yes, that is definitely better. I added dbdb0141551aa4eee7eabfc6929e53e184131379.